### PR TITLE
CLOUDP-81286: Pass git version to the binary on build

### PR DESCRIPTION
**Pull request by vasilevp**

### CODE REVIEW

File: .github/workflows/release-post-merge.yml [hunk 1]
```diff
@@ -16,61 +16,61 @@ jobs:
...
-        tar czvf atlas-operator-all-in-one-${{ steps.tag.outputs.version }}.tar.gz -C deploy all-in-one.yaml
+      - name: Push Atlas Operator to Registry
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          repository: ${{ env.DOCKER_RELEASE_REPO }}
+          registry: ${{ env.DOCKER_REGISTRY }}
+          tags: ${{ steps.tag.outputs.version }}
+          build_args: VERSION=${{ steps.tag.outputs.tag }}
```
Comment (vasilevp) [inline .github/workflows/release-post-merge.yml:48]: This is the only change here, everything else is formatting

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -11,6 +11,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0 # required for tags
```
Comment (vasilevp) [inline .github/workflows/test.yml:15]: Same here

Comment (leo-ri) [reply .github/workflows/test.yml:15]: hmmm....we don't get git tags, no?

**Review by leo-ri — APPROVED** [review]

File: main.go [hunk 3]
```diff
@@ -62,6 +69,12 @@ func main() {
 	logger := ctrzap.NewRaw(ctrzap.UseDevMode(true), ctrzap.StacktraceLevel(zap.ErrorLevel))
 
 	config := parseConfiguration(logger.Sugar())
+
+	if config.Version {
```
Comment (antonlisovenko) [inline main.go:73]: I'd suggest to always print the version on the Operator start.

File: main.go [hunk 4]
```diff
@@ -151,6 +165,7 @@ func parseConfiguration(log *zap.SugaredLogger) Config {
 	flag.BoolVar(&config.EnableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.BoolVar(&config.Version, "version", false, "Display current version and exit")
```
Comment (antonlisovenko) [inline main.go:168]: not sure I understand the value of this flag. Can we just always print the version on the Operator start?

File: pkg/controller/atlas/client.go [hunk 5]
```diff
@@ -28,7 +28,8 @@ func Client(atlasDomain string, connection Connection, log *zap.SugaredLogger) (
 	if err != nil {
 		return mongodbatlas.Client{}, err
 	}
-	client.UserAgent = userAgent
+	client.UserAgent = fmt.Sprintf("%s/%s (%s;%s)", "MongoDBAtlasKubernetesOperator", ProductVersion, runtime.GOOS, runtime.GOARCH)
```
Comment (antonlisovenko) [inline pkg/controller/atlas/client.go:31]: can we please add the unit test for this?

**Review by antonlisovenko — CHANGES_REQUESTED** [review]
Looks good 👍 
please add a simple unit test to ensure the correct user agent is sent.

**Review by antonlisovenko — APPROVED** [review]
LGTM!

**Pull request merged**